### ISSUE TITLE
Filter language is not working properly

### ISSUE THREAD

**Issue by KNTRO** [tootsuite/mastodon#11118]
### Expected behaviour
The _Local timeline_ or the _Federated timeline_ should display toots written in Esperanto only.

### Actual behaviour
Only toots with pictures are shown.

### Steps to reproduce the problem

1. Go to [_Preferences_ > _Others_](https://mastodon.social/settings/preferences/other).
2. In the _Public timeline_ section check the _Esperanto_ box only and then click on the blue _Save changes_ button.
3. Go back to the [Mastodon's _Getting started_ webpage](https://mastodon.social/web/getting-started) and select either _Local timeline_ or _Federated timeline_.

### Specifications
Mastodon Web **v2.9.0**, using Mozilla Firefox 67.0.3 64-bit, on Xubuntu 19.04, 64-bit.

![Mastodon1](https://user-images.githubusercontent.com/2568821/59748685-0461c280-9252-11e9-9349-2ec269a4b85c.jpg)
![Mastodon2](https://user-images.githubusercontent.com/2568821/59748686-04fa5900-9252-11e9-965d-3d097e685cdb.jpg)

**Comment by KNTRO** [comment]
**UPDATE**:
From time to time, a few toots with French texts appear.

**Comment by trwnh** [comment]
There may not be any posts identified as Esperanto.

Also, the French posts may simply be unidentified.

**Comment by KNTRO** [comment]
> There may not be any posts identified as Esperanto.

A quick search [using the hashtag _#Esperanto_ shows recent posts](https://mastodon.social/tags/esperanto) in that language. I mean, despite the hashtag: there are posts (simple text) in that language.

> Also, the French posts may simply be unidentified.

Exactly, because as I said, the filter language is not working properly. If I filter languages, then I should not see toots in French or any other unselected language.

**Comment by trwnh** [comment]
None of the posts I see tagged as `#esperanto` were actually recognized as Esperanto. Most of them seem to be mapped as English (although in most cases it appears to be correct, since they are English posts discussing the Esperanto language as a concept).

In general, language detection is actually harder than it may seem; see some earlier problems surrounding this issue: https://github.com/tootsuite/mastodon/pull/9760#discussion_r265834656

**Comment by KNTRO** [comment]
Oh, and the _Filter language_ option can't even work based on the user's _Posting language_ setting at [_Preferences_ > _Other_ > _Posting defaults_](https://mastodon.social/settings/preferences/other)? :thinking:

**Comment by Gargron** [comment]
Not everyone has that setting set to anything

**Comment by KNTRO** [comment]
This is my humble opinion: I think you should consider make the _Filter language_ work somehow. That "somehow" could be based on the _Posting language_ feature. Otherwise, we have 2 features (_Filter language_ and _Posting language_) that don't do a thing actually.

A side note: could those 2 options plus [_Interface language_](https://mastodon.social/settings/preferences/appearance) be on one same menu called _Language_ or something like that?

**Comment by trwnh** [comment]
I agree the filters are not as useful for non-English languages right now, but they're not useless. "Posting language" sets the default language of your own posts; "Filter languages" works as advertised but the problem is that most posts aren't explicitly identified.

Fixing the auto-detection is very hard, so I'm personally in favor of making it easier / encouraged to define language. I imagine most people don't have "Posting language" set either because they are unaware that it exists, or because they are multilingual and leave it to Auto-detect, or they have it set to English because they post primarily in English. There's definitely room for more discussion on this, though.

**Comment by jotaemei** [comment]
Well, this one may have been classified by the bot as "won't fix" due to inactivity, but another issue was opened a year later (2020) to address the bug. https://github.com/mastodon/mastodon/issues/13891

**Comment by jklawiter** [comment]
As I know if you set a language you will see posts in selected language **and also** posts with **no language set at all**.
So IMO simple solution is to filter out also toots where language is not set.
Maybe if anyone want to see it also language "not set" should be just added to languages list.
